agent/server.go

```diff
@@ -52,7 +54,8 @@ type RPCServer interface {
 // server implements RPCServer for an agent.
 type server struct {
 	*grpc.Server
-	agent *agent
+	agent       *agent
+	httpServers []*http.Server
```

Comment:
It looks like you're only saving enough to be able to call `Shutdown` and don't need anything beyond that - factor these into a slice of `gracefulCloser`?
```go
type server struct {
	// ...
	closers []gracefulCloser
}

type gracefulCloser interface {
	Shutdown(context.Context)
}
```

Reply:
server.Addr is being used when logging errors which might be useful.
Will keep it as is for now, but will look to clean it up later.